File: ios/App/View/Menu/MenuView.swift
import SwiftUI

/// Warm café palette, mirrored from MenuPage.styles.css.ts.
private enum MenuPalette {
  static let cream = Color(red: 0.980, green: 0.961, blue: 0.925)
  static let espresso = Color(red: 0.169, green: 0.129, blue: 0.102)
  static let espressoSoft = Color(red: 0.478, green: 0.416, blue: 0.361)
  static let line = Color(red: 0.902, green: 0.863, blue: 0.796)
  static let copper = Color(red: 0.706, green: 0.384, blue: 0.176)
  static let copperSoft = Color(red: 0.957, green: 0.898, blue: 0.835)
  static let open = Color(red: 0.243, green: 0.420, blue: 0.310)
  static let closed = Color(red: 0.627, green: 0.294, blue: 0.235)
}

struct MenuView: View {
  @State private var activeSection = MenuContent.menu[0].title
  @State private var filters: Set<MenuContent.Dietary> = []
  @Environment(\.openURL) private var openURL

  private var section: MenuContent.Section {
    MenuContent.menu.first { $0.title == activeSection } ?? MenuContent.menu[0]
  }

  private var visibleItems: [MenuContent.Item] {
    section.items.filter { item in filters.allSatisfy { item.dietary.contains($0) } }
  }

  var body: some View {
    ZStack {
      MenuPalette.cream.ignoresSafeArea()
      ScrollView {
        VStack(alignment: .leading, spacing: 0) {
          hero
          sectionTabs
          dietaryRow
          if let note = section.note {
            Text(note)
              .font(.footnote.italic())
              .foregroundStyle(MenuPalette.espressoSoft)
              .frame(maxWidth: .infinity)
              .padding(.top, 10)
          }
          itemsList
          hoursSection
          contactSection
          Text("\(MenuContent.name) — \(MenuContent.tagline). Built with Repobot.")
            .font(.caption)
            .foregroundStyle(MenuPalette.espressoSoft)
            .frame(maxWidth: .infinity)
            .padding(.top, 28)
        }
        .padding(.horizontal, 22)
        .padding(.bottom, 48)
      }
    }
  }

  // MARK: - Hero

  private var hero: some View {
    let now = Date()
    let calendar = Calendar.current
    let day = calendar.component(.weekday, from: now) - 1  // 1..7 -> 0..6
    let minute = calendar.component(.hour, from: now) * 60 + calendar.component(.minute, from: now)
    let status = MenuHours.statusAt(MenuContent.weeklyHours, day: day, minute: minute)
    let label = MenuHours.statusLabel(MenuContent.weeklyHours, day: day, minute: minute)

    return VStack(spacing: 10) {
      Text(MenuContent.name)
        .font(.system(size: 38, weight: .bold, design: .serif))
        .foregroundStyle(MenuPalette.espresso)
        .multilineTextAlignment(.center)
        .padding(.top, 26)
      Text(MenuContent.tagline)
        .font(.subheadline)
        .foregroundStyle(MenuPalette.espressoSoft)
      HStack(spacing: 8) {
        Circle()
          .fill(status.open ? MenuPalette.open : MenuPalette.closed)
          .frame(width: 8, height: 8)
        Text(label)
          .font(.footnote.weight(.semibold))
          .foregroundStyle(MenuPalette.espresso)
      }
      .padding(.horizontal, 16)
      .padding(.vertical, 8)
      .background(Capsule().fill(Color.white))
      .overlay(Capsule().strokeBorder(MenuPalette.line))
      Text(MenuContent.about)
        .font(.footnote)
        .foregroundStyle(MenuPalette.espressoSoft)
        .multilineTextAlignment(.center)
        .lineSpacing(3)
        .padding(.top, 6)
    }
    .frame(maxWidth: .infinity)
  }

  // MARK: - Menu

  private var sectionTabs: some View {
    ScrollView(.horizontal, showsIndicators: false) {
      HStack(spacing: 8) {
        ForEach(MenuContent.menu) { s in
          let isActive = s.title == activeSection
          Button {
            activeSection = s.title
          } label: {
            Text(s.title)
              .font(.footnote.weight(.semibold))
              .foregroundStyle(isActive ? MenuPalette.cream : MenuPalette.espressoSoft)
              .padding(.horizontal, 16)
              .padding(.vertical, 8)
              .background(Capsule().fill(isActive ? MenuPalette.copper : Color.clear))
              .overlay(
                Capsule().strokeBorder(isActive ? MenuPalette.copper : MenuPalette.line))
          }
          .buttonStyle(.plain)
        }
      }
    }
    .padding(.top, 22)
  }

  private var dietaryRow: some View {
    HStack(spacing: 8) {
      ForEach(MenuContent.Dietary.allCases, id: \.rawValue) { mark in
        let isActive = filters.contains(mark)
        Button {
          if isActive { filters.remove(mark) } else { filters.insert(mark) }
        } label: {
          Text(mark.label)
            .font(.caption.weight(.semibold))
            .foregroundStyle(isActive ? MenuPalette.copper : MenuPalette.espressoSoft)
            .padding(.horizontal, 12)
            .padding(.vertical, 5)
            .background(Capsule().fill(isActive ? MenuPalette.copperSoft : Color.clear))
            .overlay(
              Capsule()
                .strokeBorder(
                  isActive ? MenuPalette.copper : MenuPalette.line,
                  style: StrokeStyle(lineWidth: 1, dash: isActive ? [] : [4, 3])))
        }
        .buttonStyle(.plain)
      }
    }
    .frame(maxWidth: .infinity)
    .padding(.top, 12)
  }

  private var itemsList: some View {
    VStack(spacing: 0) {
      ForEach(visibleItems) { item in
        VStack(alignment: .leading, spacing: 4) {
          HStack(alignment: .firstTextBaseline) {
            HStack(spacing: 8) {
              Text(item.name)
                .font(.system(size: 17, weight: .semibold, design: .serif))
                .foregroundStyle(MenuPalette.espresso)
              if item.popular {
                Text("Popular")
                  .font(.system(size: 10, weight: .bold))
                  .foregroundStyle(MenuPalette.copper)
                  .padding(.horizontal, 8)
                  .padding(.vertical, 2)
                  .background(Capsule().fill(MenuPalette.copperSoft))
              }
              ForEach(item.dietary, id: \.rawValue) { mark in
                Text(mark.rawValue)
                  .font(.system(size: 10, weight: .bold))
                  .foregroundStyle(MenuPalette.open)
                  .padding(.horizontal, 4)
                  .padding(.vertical, 1)
                  .overlay(
                    RoundedRectangle(cornerRadius: 4).strokeBorder(MenuPalette.open))
              }
            }
            Spacer()
            Text(MenuContent.formatPrice(item.priceCents))
              .font(.system(size: 16, weight: .semibold, design: .serif))
              .foregroundStyle(MenuPalette.espresso)
          }
          Text(item.description)
            .font(.footnote)
            .foregroundStyle(MenuPalette.espressoSoft)
        }
        .padding(.vertical, 12)
        .overlay(alignment: .bottom) {
          Rectangle().fill(MenuPalette.line).frame(height: 1)
        }
      }
      if visibleItems.isEmpty {
        Text("Nothing in \(section.title.lowercased()) fits that filter — try another section.")
          .font(.footnote.italic())
          .foregroundStyle(MenuPalette.espressoSoft)
          .padding(.vertical, 28)
      }
    }
    .padding(.top, 8)
  }

  // MARK: - Hours & contact

  private var hoursSection: some View {
    let today = Calendar.current.component(.weekday, from: Date()) - 1
    return VStack(alignment: .leading, spacing: 6) {
      Text("Hours")
        .font(.system(size: 20, weight: .semibold, design: .serif))
        .foregroundStyle(MenuPalette.espresso)
        .padding(.bottom, 6)
      ForEach(0..<7, id: \.self) { d in
        let entry = MenuContent.weeklyHours.first { $0.day == d }
        let text =
          entry.map { e in
            e.intervals
              .map { "\(MenuHours.formatMinute($0.open)) – \(MenuHours.formatMinute($0.close))" }
              .joined(separator: ", ")
          } ?? "Closed"
        HStack {
          Text(MenuHours.dayNames[d])
            .foregroundStyle(d == today ? MenuPalette.espresso : MenuPalette.espressoSoft)
          Spacer()
          Text(text)
            .foregroundStyle(MenuPalette.espresso)
        }
        .font(d == today ? .footnote.weight(.bold) : .footnote)
      }
      Text(MenuContent.hoursNote)
        .font(.caption.italic())
        .foregroundStyle(MenuPalette.espressoSoft)
        .padding(.top, 8)
    }
    .padding(.top, 36)
  }

  private var contactSection: some View {
    VStack(alignment: .leading, spacing: 10) {
      Text("Find us")
        .font(.system(size: 20, weight: .semibold, design: .serif))
        .foregroundStyle(MenuPalette.espresso)
        .padding(.bottom, 2)
      Text(MenuContent.address)
        .font(.footnote)
        .foregroundStyle(MenuPalette.espresso)
      contactLink("Get directions →") {
        let query = MenuContent.mapsQuery.addingPercentEncoding(
          withAllowedCharacters: .urlQueryAllowed)
        if let url = URL(string: "https://maps.google.com/?q=\(query ?? "")") {
          openURL(url)
        }
      }
      contactLink(MenuContent.phone) {
        let digits = MenuContent.phone.filter { $0.isNumber || $0 == "+" }
        if let url = URL(string: "tel:\(digits)") { openURL(url) }
      }
      contactLink(MenuContent.email) {
        if let url = URL(string: "mailto:\(MenuContent.email)") { openURL(url) }
      }
      contactLink("Instagram →") {
        if let url = URL(string: MenuContent.instagram) { openURL(url) }
      }
    }
    .padding(.top, 30)
  }

  private func contactLink(_ label: String, action: @escaping () -> Void) -> some View {
    Button(action: action) {
      Text(label)
        .font(.footnote.weight(.semibold))
        .foregroundStyle(MenuPalette.copper)
    }
    .buttonStyle(.plain)
  }
}

#Preview {
  MenuView()
}

File: android/app/src/main/kotlin/com/baseapp/android/view/menu/MenuView.kt
package com.baseapp.android.view.menu

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import java.util.Calendar

/** Warm café palette, mirrored from MenuPage.styles.css.ts. */
private object MenuPalette {
    val cream = Color(0xFFFAF5EC)
    val espresso = Color(0xFF2B211A)
    val espressoSoft = Color(0xFF7A6A5C)
    val line = Color(0xFFE6DCCB)
    val copper = Color(0xFFB4622D)
    val copperSoft = Color(0xFFF4E5D5)
    val open = Color(0xFF3E6B4F)
    val closed = Color(0xFFA04B3C)
}

@Composable
fun MenuView() {
    var activeSection by remember { mutableStateOf(MenuContent.menu.first().title) }
    var filters by remember { mutableStateOf(setOf<MenuContent.Dietary>()) }
    val context = LocalContext.current

    val section = MenuContent.menu.firstOrNull { it.title == activeSection }
        ?: MenuContent.menu.first()
    val visibleItems = section.items.filter { item -> filters.all { it in item.dietary } }

    fun open(uri: String) {
        context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(uri)))
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(MenuPalette.cream)
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 22.dp)
            .padding(bottom = 48.dp),
    ) {
        Hero()
        SectionTabs(activeSection) { activeSection = it }
        DietaryRow(filters) { mark ->
            filters = if (mark in filters) filters - mark else filters + mark
        }
        section.note?.let { note ->
            Text(
                text = note,
                fontSize = 13.sp,
                fontStyle = FontStyle.Italic,
                color = MenuPalette.espressoSoft,
                textAlign = TextAlign.Center,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = 10.dp),
            )
        }
        visibleItems.forEach { item -> ItemRow(item) }
        if (visibleItems.isEmpty()) {
            Text(
                text = "Nothing in ${section.title.lowercase()} fits that filter — try another section.",
                fontSize = 13.sp,
                fontStyle = FontStyle.Italic,
                color = MenuPalette.espressoSoft,
                textAlign = TextAlign.Center,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(vertical = 28.dp),
            )
        }
        HoursSection()
        ContactSection(::open)
        Text(
            text = "${MenuContent.NAME} — ${MenuContent.TAGLINE}. Built with Repobot.",
            fontSize = 12.sp,
            color = MenuPalette.espressoSoft,
            textAlign = TextAlign.Center,
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 28.dp),
        )
    }
}

@Composable
private fun Hero() {
    val now = Calendar.getInstance()
    val day = now.get(Calendar.DAY_OF_WEEK) - 1 // 1..7 -> 0..6
    val minute = now.get(Calendar.HOUR_OF_DAY) * 60 + now.get(Calendar.MINUTE)
    val status = MenuHours.statusAt(MenuContent.weeklyHours, day, minute)
    val label = MenuHours.statusLabel(MenuContent.weeklyHours, day, minute)

    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        modifier = Modifier
            .fillMaxWidth()
            .padding(top = 40.dp),
    ) {
        Text(
            text = MenuContent.NAME,
            fontSize = 36.sp,
            fontWeight = FontWeight.Bold,
            fontFamily = FontFamily.Serif,
            color = MenuPalette.espresso,
            textAlign = TextAlign.Center,
        )
        Text(
            text = MenuContent.TAGLINE,
            fontSize = 15.sp,
            color = MenuPalette.espressoSoft,
            modifier = Modifier.padding(top = 6.dp),
        )
        Row(
            verticalAlignment = Alignment.CenterVertically,
            modifier = Modifier
                .padding(top = 14.dp)
                .clip(RoundedCornerShape(999.dp))
                .background(Color.White)
                .border(1.dp, MenuPalette.line, RoundedCornerShape(999.dp))
                .padding(horizontal = 16.dp, vertical = 8.dp),
        ) {
            Box(
                modifier = Modifier
                    .size(8.dp)
                    .clip(CircleShape)
                    .background(if (status.open) MenuPalette.open else MenuPalette.closed),
            )
            Text(
                text = label,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                color = MenuPalette.espresso,
                modifier = Modifier.padding(start = 8.dp),
            )
        }
        Text(
            text = MenuContent.ABOUT,
            fontSize = 13.sp,
            lineHeight = 20.sp,
            color = MenuPalette.espressoSoft,
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(top = 14.dp),
        )
    }
}

@Composable
private fun SectionTabs(active: String, onSelect: (String) -> Unit) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        modifier = Modifier
            .horizontalScroll(rememberScrollState())
            .padding(top = 22.dp),
    ) {
        MenuContent.menu.forEach { section ->
            val isActive = section.title == active
            Text(
                text = section.title,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                color = if (isActive) MenuPalette.cream else MenuPalette.espressoSoft,
                modifier = Modifier
                    .clip(RoundedCornerShape(999.dp))
                    .background(if (isActive) MenuPalette.copper else Color.Transparent)
                    .border(
                        1.dp,
                        if (isActive) MenuPalette.copper else MenuPalette.line,
                        RoundedCornerShape(999.dp),
                    )
                    .clickable { onSelect(section.title) }
                    .padding(horizontal = 16.dp, vertical = 8.dp),
            )
        }
    }
}

@Composable
private fun DietaryRow(filters: Set<MenuContent.Dietary>, onToggle: (MenuContent.Dietary) -> Unit) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterHorizontally),
        modifier = Modifier
            .fillMaxWidth()
            .padding(top = 12.dp),
    ) {
        MenuContent.Dietary.entries.forEach { mark ->
            val isActive = mark in filters
            Text(
                text = mark.label,
                fontSize = 12.sp,
                fontWeight = FontWeight.SemiBold,
                color = if (isActive) MenuPalette.copper else MenuPalette.espressoSoft,
                modifier = Modifier
                    .clip(RoundedCornerShape(999.dp))
                    .background(if (isActive) MenuPalette.copperSoft else Color.Transparent)
                    .border(
                        1.dp,
                        if (isActive) MenuPalette.copper else MenuPalette.line,
                        RoundedCornerShape(999.dp),
                    )
                    .clickable { onToggle(mark) }
                    .padding(horizontal = 12.dp, vertical = 5.dp),
            )
        }
    }
}

@Composable
private fun ItemRow(item: MenuContent.Item) {
    Column(modifier = Modifier.padding(top = 14.dp)) {
        Row(verticalAlignment = Alignment.Top) {
            Column(modifier = Modifier.weight(1f)) {
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(8.dp),
                ) {
                    Text(
                        text = item.name,
                        fontSize = 17.sp,
                        fontWeight = FontWeight.SemiBold,
                        fontFamily = FontFamily.Serif,
                        color = MenuPalette.espresso,
                    )
                    if (item.popular) {
                        Text(
                            text = "Popular",
                            fontSize = 10.sp,
                            fontWeight = FontWeight.Bold,
                            color = MenuPalette.copper,
                            modifier = Modifier
                                .clip(RoundedCornerShape(999.dp))
                                .background(MenuPalette.copperSoft)
                                .padding(horizontal = 8.dp, vertical = 2.dp),
                        )
                    }
                    item.dietary.forEach { mark ->
                        Text(
                            text = mark.mark,
                            fontSize = 10.sp,
                            fontWeight = FontWeight.Bold,
                            color = MenuPalette.open,
                            modifier = Modifier
                                .border(1.dp, MenuPalette.open, RoundedCornerShape(4.dp))
                                .padding(horizontal = 4.dp),
                        )
                    }
                }
                Text(
                    text = item.description,
                    fontSize = 13.sp,
                    lineHeight = 19.sp,
                    color = MenuPalette.espressoSoft,
                    modifier = Modifier.padding(top = 3.dp),
                )
            }
            Text(
                text = MenuContent.formatPrice(item.priceCents),
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                fontFamily = FontFamily.Serif,
                color = MenuPalette.espresso,
                modifier = Modifier.padding(start = 12.dp),
            )
        }
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 12.dp)
                .height(1.dp)
                .background(MenuPalette.line),
        )
    }
}

@Composable
private fun HoursSection() {
    val today = Calendar.getInstance().get(Calendar.DAY_OF_WEEK) - 1
    Column(modifier = Modifier.padding(top = 36.dp)) {
        Text(
            text = "Hours",
            fontSize = 20.sp,
            fontWeight = FontWeight.SemiBold,
            fontFamily = FontFamily.Serif,
            color = MenuPalette.espresso,
            modifier = Modifier.padding(bottom = 8.dp),
        )
        MenuHours.dayNames.forEachIndexed { d, name ->
            val entry = MenuContent.weeklyHours.firstOrNull { it.day == d }
            val text = entry?.intervals?.joinToString(", ") { (open, close) ->
                "${MenuHours.formatMinute(open)} – ${MenuHours.formatMinute(close)}"
            } ?: "Closed"
            val weight = if (d == today) FontWeight.Bold else FontWeight.Normal
            Row(modifier = Modifier.padding(vertical = 3.dp)) {
                Text(
                    text = name,
                    fontSize = 14.sp,
                    fontWeight = weight,
                    color = if (d == today) MenuPalette.espresso else MenuPalette.espressoSoft,
                )
                Spacer(modifier = Modifier.weight(1f))
                Text(
                    text = text,
                    fontSize = 14.sp,
                    fontWeight = weight,
                    color = MenuPalette.espresso,
                )
            }
        }
        Text(
            text = MenuContent.HOURS_NOTE,
            fontSize = 12.sp,
            fontStyle = FontStyle.Italic,
            color = MenuPalette.espressoSoft,
            modifier = Modifier.padding(top = 8.dp),
        )
    }
}

@Composable
private fun ContactSection(open: (String) -> Unit) {
    Column(
        verticalArrangement = Arrangement.spacedBy(10.dp),
        modifier = Modifier.padding(top = 30.dp),
    ) {
        Text(
            text = "Find us",
            fontSize = 20.sp,
            fontWeight = FontWeight.SemiBold,
            fontFamily = FontFamily.Serif,
            color = MenuPalette.espresso,
        )
        Text(text = MenuContent.ADDRESS, fontSize = 13.sp, color = MenuPalette.espresso)
        ContactLink("Get directions →") {
            open("https://maps.google.com/?q=${Uri.encode(MenuContent.MAPS_QUERY)}")
        }
        ContactLink(MenuContent.PHONE) {
            open("tel:${MenuContent.PHONE.filter { it.isDigit() || it == '+' }}")
        }
        ContactLink(MenuContent.EMAIL) { open("mailto:${MenuContent.EMAIL}") }
        ContactLink("Instagram →") { open(MenuContent.INSTAGRAM) }
    }
}

@Composable
private fun ContactLink(label: String, onClick: () -> Unit) {
    Text(
        text = label,
        fontSize = 13.sp,
        fontWeight = FontWeight.SemiBold,
        color = MenuPalette.copper,
        modifier = Modifier.clickable(onClick = onClick),
    )
}
